### DISCUSSION TITLE
How to use GraphQL tokens behind basic auth?

### DISCUSSION THREAD

**Discussion by gopeter** [craftcms/cms#10539] [Q&A] [answered]
I'm running Craft CMS behind Nginx where I've configured basic authentication for my staging setup. Now I want to use private GraphQL schemas, but my frontend can't send the basic auth credentials and the GraphQL token. Is there a way to customize the `Authorization` header?

**Comment by brandonkelly** [comment]
You can pass both the `Basic` and `Bearer` tokens together in the same `Authorization` header:

```
Authorization: Basic XXXXXXXXXXXXXXXX, Bearer XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```

**Reply by gopeter** [reply]
But this is [against the RFC](https://tools.ietf.org/html/rfc7235#appendix-C). [Some users also mentioned](https://stackoverflow.com/questions/29282578/multiple-http-authorization-headers) that it does not work with Apache. That's why I thought that it would be better to allow users to customize the header :-)

**Reply by brandonkelly** [reply]
Ah ok, didn’t know that!

**Comment by brandonkelly** [comment, marked as answer]
Just added support for passing the bearer token via an `X-Craft-Authorization` header (0fce7a745fe421d7405fcd72f005f3b9a6358785).

**Reply by gopeter** [reply]
Thanks a lot 😊

**Reply by brandonkelly** [reply]
Craft 3.7.33 is out now with this change.

**Reply by janhenckens** [reply]
Well, today I learned something. Was struggling with the same issue on our staging setup (which runs apache), using `X-Craft-Authorization` fixed it 👍
